src/plugins/data/public/index_patterns/fields/field.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+// @ts-ignore
+import { ObjDefine } from './obj_define';
+// @ts-ignore
+import { shortenDottedString } from '../../../../../legacy/core_plugins/kibana/common/utils/shorten_dotted_string';
+import { IndexPattern } from '../index_patterns';
+import { getNotifications, getFieldFormats } from '../services';
+import { IFieldType, getKbnFieldType, IFieldSubType, FieldFormat } from '../../../common';
+
+export type FieldSpec = Record<string, any>;
+
+/** @deprecated
+ *  Please use IFieldType instead
+ * */
+export type FieldType = IFieldType;
```

Comment:
This is not needed in NP.
You can keep this deprecated export only in old platform

Reply:
Done